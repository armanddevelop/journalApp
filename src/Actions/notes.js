import { db } from "../FireBase/fireBaseConfig";
import { types } from "../Types/types";

export const notesActiveAction = (id, note) => ({
  type: types.notesActive,
  payload: {
    id,
    ...note,
  },
});

export const startNewNoteAction = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const newNote = {
      title: "",
      body: "",
      date: new Date().getTime(),
    };
    const docRef = await db.collection(`${uid}/jornal/notes`).add(newNote);
    dispatch(notesActiveAction(docRef.id, newNote));
  };
};
