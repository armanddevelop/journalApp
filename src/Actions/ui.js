import { types } from "../Types/types";

export const setErrorAction = (error) => ({
  type: types.uiSetError,
  payload: error,
});

export const unSetErrorAction = () => ({
  type: types.uiRemoveError,
});

export const startLoading = () => ({
  type: types.uiStartLoading,
});

export const finishLoading = () => ({
  type: types.uiFinishLoading,
});
