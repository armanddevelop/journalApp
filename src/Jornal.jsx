import { AppRouter } from "../src/Routers/AppRouter";

export const Jornal = () => {
  return (
    <div>
      <AppRouter />
    </div>
  );
};
