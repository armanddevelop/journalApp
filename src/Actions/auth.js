import { types } from "../Types/types";
import { firebase, googleAuthProvider } from "../FireBase/fireBaseConfig";
import {
  finishLoading,
  startLoading,
  errorInFirebase,
  removeErrorInFirebase,
} from "./ui";
import { notesLogOutAction } from "./notes";

export const logInAction = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});

export const logOutAction = () => ({
  type: types.logout,
});

export const startLogOut = () => {
  return async (dispatch) => {
    try {
      await firebase.auth().signOut();
      dispatch(logOutAction());
      dispatch(notesLogOutAction());
    } catch (error) {
      console.log("shit happen in startLogOut ", error);
    }
  };
};

export const getGoogleLogInAction = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
      });
  };
};

export const getLoginUserAction = (email, password) => {
  return (dispatch) => {
    dispatch(startLoading());
    dispatch(removeErrorInFirebase());
    firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
        dispatch(finishLoading());
      })
      .catch((error) => {
        dispatch(finishLoading());
        dispatch(errorInFirebase(error.message));
      });
  };
};

export const registerUserAction = (name, email, password) => {
  return (dispatch) => {
    dispatch(startLoading());
    dispatch(removeErrorInFirebase());
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        const { uid, displayName } = user;
        dispatch(logInAction(uid, displayName));
        dispatch(finishLoading());
      })
      .catch((error) => {
        dispatch(finishLoading());
        dispatch(errorInFirebase(error.message));
      });
  };
};
