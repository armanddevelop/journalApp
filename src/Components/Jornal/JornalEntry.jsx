import moment from "moment";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import { notesActiveAction } from "../../Actions/notes";

export const JornalEntry = ({ title, body, date, url, id }) => {
  const noteDate = moment(date);
  const dispatch = useDispatch();
  const handleEntryClick = () => {
    const noteObj = { title, body, date, url };
    dispatch(notesActiveAction(id, noteObj));
  };
  return (
    <div className="jornal__entry pointer" onClick={handleEntryClick}>
      {url && (
        <div
          className="jornal__entry-picture"
          style={{
            backgroundSize: "cover",
            backgroundImage: `url(${url})`,
          }}
        ></div>
      )}
      <div className="jornal__entry-body">
        <p className="jornal__entry-title">{title}</p>
        <p className="jornal__entry-content">{body}</p>
      </div>
      <div className="jornal__entry-date-box">
        <span>{noteDate.format("dddd")}</span>
        <h4>{noteDate.format("Do")}</h4>
      </div>
    </div>
  );
};

JornalEntry.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  date: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  url: PropTypes.string,
};
