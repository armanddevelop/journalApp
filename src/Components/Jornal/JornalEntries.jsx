import { JornalEntry } from "./JornalEntry";

export const JornalEntries = () => {
  const entries = [1, 2, 3, 4, 5];
  return (
    <div className="jornal__entries">
      {entries.map((entry) => (
        <JornalEntry key={entry} />
      ))}
    </div>
  );
};
