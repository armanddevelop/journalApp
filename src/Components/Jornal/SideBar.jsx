import { useDispatch, useSelector } from "react-redux";
import { startLogOut } from "../../Actions/auth";
import { startNewNoteAction } from "../../Actions/notes";
import { JornalEntries } from "./JornalEntries";

export const SideBar = () => {
  const dispatch = useDispatch();
  const { name } = useSelector((state) => state.auth);
  const handleLogOut = () => {
    dispatch(startLogOut());
  };
  const handleNewEntry = () => {
    console.log("licha");
    dispatch(startNewNoteAction());
  };
  return (
    <aside className="jornal__sidebar">
      <div className="jornal__sidebar-navbar">
        <h3 className="mt-5">
          <i className="far fa-moon"></i>
          <span> {name}</span>
        </h3>
        <button className="btn" onClick={handleLogOut}>
          LogOut
        </button>
      </div>
      <div className="jornal__new-entry" onClick={handleNewEntry}>
        <i className="far fa-calendar-plus fa-5x jornal__new-entry-icon"></i>
        <p className="mt-5">New Entry</p>
      </div>
      <JornalEntries />
    </aside>
  );
};
