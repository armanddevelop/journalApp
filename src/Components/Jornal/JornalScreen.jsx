import { SelectOption } from "./SelectOption";
import { useSelector } from "react-redux";
import { NoteScreen } from "../Notes/NoteScreen";
import { SideBar } from "./SideBar";
export const JornalScreen = () => {
  const { active } = useSelector((state) => state.notes);
  return (
    <div className="jornal__main-content">
      <SideBar />
      <main>{!active ? <SelectOption /> : <NoteScreen />}</main>
    </div>
  );
};
