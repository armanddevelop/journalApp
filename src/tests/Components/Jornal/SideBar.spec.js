import { mount } from "enzyme";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { startLogOut } from "../../../Actions/auth";
import { startNewNoteAction } from "../../../Actions/notes";
import { SideBar } from "../../../Components/Jornal/SideBar";

jest.mock("../../../Actions/auth", () => ({
  startLogOut: jest.fn(),
}));
jest.mock("../../../Actions/notes", () => ({
  startNewNoteAction: jest.fn(),
}));
const initState = {
  auth: {
    name: "Alicha",
  },
  ui: {
    errorMsgFb: {
      error: false,
      msgError: "",
    },
  },
  notes: {
    active: {
      id: "safdbdg32435",
    },
    notes: [],
    saveNote: {
      isSave: true,
      message: "All rigth",
      title: "note save",
    },
  },
};
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
let store = mockStore(initState);
store.dispatch = jest.fn();
const wrapper = mount(
  <Provider store={store}>
    <SideBar />
  </Provider>
);
describe("Test in SideBar component", () => {
  it("SnapShot to SideBar component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Should show the right name in the component", () => {
    const name = wrapper.find("span").text();
    const { auth } = store.getState((state) => state);

    expect(name.trim()).toBe(auth.name);
  });

  it("Should execute handleLogOut", () => {
    wrapper.find(".btn").prop("onClick")();

    expect(startLogOut).toBeCalled();
  });

  it("Should execute startNewNoteAction", () => {
    wrapper.find(".jornal__new-entry").prop("onClick")();

    expect(startNewNoteAction).toHaveBeenCalled();
  });
});
