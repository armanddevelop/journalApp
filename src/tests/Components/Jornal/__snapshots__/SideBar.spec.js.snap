// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in SideBar component SnapShot to SideBar component 1`] = `
<aside
  className="jornal__sidebar"
>
  <div
    className="jornal__sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <i
        className="far fa-moon"
      />
      <span>
         
        Alicha
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      LogOut
    </button>
  </div>
  <div
    className="jornal__new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x jornal__new-entry-icon"
    />
    <p
      className="mt-5"
    >
      New Entry
    </p>
  </div>
  <div
    className="jornal__entries"
  />
</aside>
`;
